lib/saml/de_serializer.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+module SAML
+  class DeSerializer
+
+    ## saml_response = OneLogin::RubySaml::Response
+    def initialize(saml_response)
+      @saml_response = saml_response
+    end
+
+    def user_attributes
```

Comment:
I think all of this can be moved into the initialize

and you just call the class UserAttributes